inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,11 +102,19 @@ def __init__(self) -> None:
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
```

Comment:
Add a unit test to verify that the `add` method behaves correctly when the GPG key URI is not in the trusted list. [important]

Reply:
@tsirlapu this does need a new unit test

Reply:
@tsirlapu oh, i see one is added, never mind!